src/api/controllers/user.js

```diff
@@ -0,0 +1,66 @@
+const User = require("../../models/User");
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+
+//Signup
+const signup = async (req, res) => {
+  try {
+    const { username, firstname, lastname, email, password } = req.body;
+    const data = {
+      username,
+      email,
+      firstname,
+      lastname,
+      password: await bcrypt.hash(password, 10),
+    };
+    const user = await User.create(data);
+    return res.status(201).json({
+      message: "User created successfully",
+      user: user,
+    });
```

Comment:
There should be consistency on the status codes, if you are returning status code 200 for successful requests then continue with it.